crypto/err/err.c

```diff
@@ -224,24 +224,33 @@ static void build_SYS_str_reasons(void)
         str->error = ERR_PACK(ERR_LIB_SYS, 0, i);
         if (str->string == NULL) {
             if (openssl_strerror_r(i, cur, sizeof(strerror_pool) - cnt)) {
-                size_t l = strlen(cur);
+                size_t l = strlen(cur) + 1;
 
                 str->string = cur;
                 cnt += l;
-                if (cnt >= sizeof(strerror_pool))
-                    cnt = sizeof(strerror_pool) - 1;
-                cur += l;
+                /*
+                 * If we have used up all the space in strerror_pool,
+                 * openssl_strerror_r didn't copy any data to cur,
```

Comment:
If `openssl_strerror_r` didn't copy anything, then `strlen(cur)` about should have given zero, right? Can't that be detected earlier and avoid `cnt += l`?

Reply:
When ```cnt``` reaches  sizeof(strerror_pool), we could even skip openssl_strerror_r() and the related handling altogether to simplify the logic.